jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/IteratingCallback.java

```diff
@@ -41,20 +44,27 @@
  * <p>
  * Subclasses must implement method {@link #process()} where the
  * asynchronous sub-task is initiated and a suitable {@link Action}
- * is returned to this callback to indicate the overall progress of
+ * is returned to this callback to indicate the overall progress ofk
  * the large asynchronous task.
  * This callback is passed to the asynchronous sub-task, and a call
  * to {@link #succeeded()} on this callback represents the successful
  * completion of the asynchronous sub-task, while a call to
  * {@link #failed(Throwable)} on this callback represents the
  * completion with a failure of the large asynchronous task.
+ * <p>
+ * For most purposes, the {@link #succeeded()} and {@link #failed(Throwable)}
```

Comment:
Don't we want to make them `final` for 12.1? I really see no reason to leave a dangerous possibility open.

Plus, `AsyncProxyServlet.StreamReader` overrides `failed()` which gives a compilation error as `succeeded` and `failed` are marked final.

Reply:
They are already final... just old javadoc.